core/config/project_settings.cpp

```diff
@@ -39,6 +39,10 @@
 #include "core/io/file_access_pack.h"
 #include "core/io/marshalls.h"
 #include "core/os/keyboard.h"
+#include "core/os/thread_safe.h"
+#include "core/templates/hash_map.h"
+#include "core/templates/local_vector.h"
+#include "core/templates/rb_set.h"
```

Comment:
These should all be removed, they're already included via `class_db.h`, as otherwise things wouldn't compile, as they're used in the header

Reply:
I tried to compile without them before and there were errors
![image](https://github.com/godotengine/godot/assets/116561933/b589bee7-2c45-4d3f-92c8-157fc94c6289)

Reply:
They're all used in the header, so that would break anyway if they weren't included indirectly, try adding back `rb_set.h` to the header, it shouldn't be moved in either case, either deleted or kept

Reply:
No, it compiles and passes all the tests. These includes are unnecessary in the header

Reply:
No they aren't, they aren't indirectly included, they are accidentally included in any other context, otherwise it would compile, it wouldn't be possible for it to only fail in the cpp file